packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -0,0 +1,117 @@
+import type { RestrictedControllerMessenger } from '@metamask/base-controller';
+import { BaseControllerV2 } from '@metamask/base-controller';
+
+const controllerName = 'QueuedRequestController';
+
+export const QueuedRequestControllerActionTypes = {
+  enqueueRequest: `${controllerName}:enqueueRequest` as const,
+};
+
+export const QueuedRequestControllerEventTypes = {
+  countChanged: `${controllerName}:countChanged` as const,
+};
+
+export type QueuedRequestControllerState = {
+  [k: string]: any;
+};
+
+export type QueuedRequestControllerCountChangedEvent = {
+  type: typeof QueuedRequestControllerEventTypes.countChanged;
+  payload: [number];
+};
```

Comment:
Just curious. How will this be used?

Reply:
https://github.com/MetaMask/metamask-extension/pull/21037/files#diff-215d2cf6fb64ee62b9f484e7bbdd33246739e39b854e3cc8c2245b026352c12aR778

Its used to updaate the number of pending requests that is show in the extension icon